src/modules/wreck/wrexecd.c

```diff
@@ -2382,24 +2383,24 @@ int main (int ac, char **av)
     if (!prog_ctx_getopt (ctx, "no-pmi-server") && prog_ctx_initialize_pmi (ctx) < 0)
         wlog_fatal (ctx, 1, "failed to initialize pmi-server");
 
-    if (exec_commands (ctx) == 0) {
+    exec_rc = exec_commands (ctx);
 
-        if (flux_reactor_run (flux_get_reactor (ctx->flux), 0) < 0)
-            wlog_err (ctx, "flux_reactor_run: %s", flux_strerror (errno));
-
-        rexec_state_change (ctx, "complete");
-        wlog_msg (ctx, "job complete. exiting...");
-
-        lua_stack_call (ctx->lua_stack, "rexecd_exit");
-    }
+    if (!exec_rc && flux_reactor_run (flux_get_reactor (ctx->flux), 0) < 0)
```

Comment:
For some reason `if (!exec_rc ..` confused me for a second here, would you be willing to change to `if (exec_rc == 0 &&` for clarity? Might just be me though...